### PR TITLE
feat(automata): compute exact transition-Parikh profiles

### LINKED ISSUE #2301: [Operation]: Compute exact transition-Parikh profiles for bounded automaton paths
## Parent gap / RFC

This is a focused child of the finite-automaton domain introduced by #1854 and adjacent to the word/factor Parikh vocabulary in #1905.

Atlas's Fibonacci macro certificates enumerate exact transition/port-use vectors for bounded path lengths and then feed those vectors unchanged into exact linear optimization. The current script is exact only through length 4 and silently samples beyond that boundary; see [`route_enum.py`](https://github.com/techno-optimist/erdos-frontier-atlas/blob/0394e3d3b249439ffabec7d96a3311aa441651b8/certificates/fibonacci-macro-residual/route_enum.py).

## Public mathematical postcondition

For a bounded finite labeled automaton, source state, target state, and exact path length, return the complete exact profile of transition-usage vectors, with the exact number of paths realizing each vector.

## Proposed operation ID

`automaton.path.transition_parikh_profile.compute`

## Input

- a canonical finite automaton with stable transition identifiers;
- source and target states;
- an exact nonnegative path length;
- the complete ordered transition axis owned by the automaton.

The public contract should count transition uses directly. Caller-defined callbacks or theorem-specific port weights do not belong in the operation.

## Result

A source-bound canonical map:

```text
transition-count vector -> exact path multiplicity
```

Optionally return one deterministic replayable witness path for each profile cell within a separately bounded witness budget.

## Completeness semantics

Every accepted request returns the complete profile for exactly the requested length and endpoints. Admission must derive state/path/count/profile/output bounds and reject excess. It must never switch from exact enumeration to sampling while retaining the same result type.

## Bounds and cost class

Preflight:

- states, transitions, alphabet, and exact length;
- path-DP states and predicted profile cells;
- coordinate/count digits;
- witness-path and serialized-result bytes.

The count-only accepted-word operation may admit larger lengths; this profile operation has a different output envelope.

## Maintained implementation

A bounded dynamic program whose state includes the automaton state and transition-count vector is a direct exact kernel. Equivalent generating-function or sparse polynomial DP implementations are private choices.

## Defining invariant

For every returned vector `v`, each coordinate is nonnegative, the coordinate sum is the path length, and the multiplicity equals the number of source-to-target paths whose transition ledger is exactly `v`. Summing all profile multiplicities must equal the existing exact-length source-to-target path count.

Any witness path must follow source transitions, end at the requested target, have the requested length, and reconstruct its vector.

## Evidence plan

- length 0 with source = target gives the zero vector once;
- length 0 with distinct endpoints gives an empty profile;
- no-path and unreachable-state cases;
- two distinct paths with the same vector distinguish set from histogram semantics;
- summing cells agrees with independent exact-length path counting;
- projecting transition counts to alphabet counts agrees with brute-force word Parikh enumeration;
- transition/state relabeling transports profiles canonically;
- mutate source, target, transition, length, count, vector, or witness and reject;
- Atlas length-3 product-state fixtures, including the reachability distinction where all clocks allow 9 anchor ports but reachable clocks allow only 7;
- a request just beyond each derived budget rejects rather than samples.

## Why this is not caller-side composition

Exact path counting loses the distribution of transition uses. Enumerating paths caller-side defeats the operation's bounded exact profile postcondition, while word-level Parikh vectors cannot distinguish distinct transitions carrying the same symbol.

## Deferred operation family

Caller-specific signed port maps, LP assembly, routability decisions, sampled research search, arbitrary additive transition weights, asymptotic growth, and symbolic-dynamics entropy remain separate composition or owned domains.

### PR DESCRIPTION
**Pull request by morluto**
## Problem

Closes #2301.

The existing regular-language operations can count fixed-length words, but a scalar count loses how often each identified transition is used. Word-level Parikh vectors also merge distinct transitions that carry the same symbol. That leaves callers unable to transport the exact transition ledger needed by downstream optimization, and the motivating Atlas script falls back to sampling beyond its small exact range.

## Solution

Add `automaton.path.transition_parikh_profile.compute`. For one finite labeled automaton, source, target, and exact length, it returns the complete lexicographically ordered histogram

```text
transition-count vector -> exact path multiplicity
```

The new canonical `FiniteLabeledAutomaton` gives every edge a stable coordinate, including parallel edges. The result retains that carrier and the complete path scope, and its model validator replays the bounded recurrence so a serialized result cannot change a transition, endpoint, length, vector, or multiplicity while preserving only local shape checks. A native `transition_parikh_profile` function and deterministic DFA-to-carrier projection expose the same mathematical values without transport-specific types.

Admission combines two independent bounds: reachable path-extension counting on the supplied carrier and a stars-and-bars bound on possible transition-count vectors. Their sharper bound controls sparse-DP updates, dense-vector work, peak live coordinates, profile cells, multiplicity digits, and projected wire bytes before expansion. Accepted requests always return the complete profile; requests outside that envelope reject rather than sample.

No dependency is added. The kernel is the defining sparse recurrence over exact integers, keyed by `(state, transition-count vector)`. Existing DFA backends do not own the required stable parallel-edge axis or source-bound transition histogram, so an adapter would add machinery without replacing the bounded mathematical kernel.

## Testing

- `make test-math TESTS=tests/math/regular_languages` — 43 passed
- `uv run --locked pytest -q tests/integration/catalog/test_public_operation_contract_conformance.py` — 484 passed
- `make check` — Ruff, formatting, mypy over 764 source files, and 2,671 tests passed

The owner lane includes brute-force and property comparisons, recurrence and projection invariants, source/result mutation rejection, and boundary cases for every derived budget. The pinned Atlas fixture exercises the actual `clock^3 × carry` carrier: 81 states, 729 identified transitions, 141 result cells, and a 253,124-byte exact result.

## Public contract impact

Adds the public operation `automaton.path.transition_parikh_profile.compute`, its request/result schemas, the canonical `FiniteLabeledAutomaton`, `AutomatonTransition`, `TransitionParikhCell`, and `TransitionParikhProfile` values, plus native `transition_parikh_profile` and `dfa_transition_carrier` functions.

The operation has exact-only semantics. Its result is the complete endpoint-bound profile for every accepted request; there is no partial or unknown result shape.

## Public operation admission

- Concrete gap: fixed-length path counts and word Parikh data do not retain the complete distribution on identified transitions.
- Why existing operations or typed values are insufficient: `regular_language.count.compute` returns one scalar, while `DFA` cannot represent partial, nondeterministic, or parallel identified transitions as a reusable coordinate carrier.
- Stable mathematical result: the complete exact map from dense transition-use vectors to positive path multiplicities for one source, target, and exact path length.
- Semantic domain and admitted execution envelope: finite partial or nondeterministic labeled automata with a contiguous transition-ID axis; requests are admitted only when complete recurrence work, intermediates, integer digits, and output fit the declared envelope.
- Controlling work, intermediate, memory, and output quantities: reachable path extensions, composition-cell bounds, sparse DP updates, transition-vector update work, peak live vector coordinates, target profile cells, multiplicity digits, and projected canonical-wire bytes.
- Exact representation, algorithm regimes, and maintained backend: dense nonnegative transition vectors and canonical integer multiplicities, computed by one bounded sparse exact recurrence. Reachability-sensitive and composition-sensitive preflights use whichever proof gives the sharper safe bound. No external backend is needed for this direct finite recurrence.
- Remaining fixed caps and their classification: 4,096 states and transitions and 32 symbols are conservative carrier-materialization bounds; length 1,000,000 is a pure-Python layer fallback. Separate operational limits cap two million DP updates, twenty million vector-coordinate updates, four million live vector coordinates, 43,000 result cells, 32,768 count digits, and a four MiB projected result. Derived work and result bounds normally decide admission before the coarse fallbacks.
- Admission decision: `KEEP` in the regular-language owner.

## Closure matrix

None. This PR admits one operation. Caller-defined transition weights, optimization, sampling, witnesses, and asymptotic analysis remain outside its postcondition.

## Checklist

- [x] `make check` passes
- [x] Explicitly relevant specialist validation is listed above (catalog conformance and the regular-language owner lane)
- [x] Harbor task or verifier changes ran the required validation (not applicable; no Harbor files changed)
- [x] Public operation changes include an owner-local admission decision
- [x] Result semantics distinguish exact, approximate, incomplete, unknown, and unavailable outcomes where applicable (exact-only; out-of-envelope requests reject)
- [x] New or changed bounds include boundary, algorithm-crossover, and realistic source-backed scale evidence
- [x] New shared abstractions replace duplication in at least two surviving production paths (not applicable; the new canonical carrier is shared by the native operation and DFA projection without replacing an existing duplicate)

### CODE REVIEW

**Comment by morluto** [comment]
@codex

File: tests/math/regular_languages/test_transition_parikh_profile.py [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+            automaton=automaton,
+            source_state=0,
+            target_state=0,
+            path_length=10,
+        )
+
+
+def test_request_rejects_excessive_intermediate_vector_coordinates() -> None:
+    automaton = _loop_automaton(159, state_count=2)
+    with pytest.raises(ValidationError, match="intermediate vector-coordinate bound"):
```
[the remark this answers is not in the document]
Comment (morluto) [reply tests/math/regular_languages/test_transition_parikh_profile.py:510]: Root cause: the review's arithmetic drops `max_layer_paths` from the `layer_cell_bound` minimum in `_require_transition_profile_envelope` (src/jacobian/math/regular_languages/operations.py). The implementation charges `min(max_layer_paths, state_count * profile_composition_bound, dp_update_bound + 1) * transition_count` intermediate coordinates, and for this loop family at path length 2 the binding term is `max_layer_paths = m^2` for every m >= 1 (since m^2 <= m(m+1) and m^2 <= m^2 + 1), so the charged count is exactly `m^3`, not `C(m+1, 2) * m`. Concretely at m=159: dp_update_bound = 25,440 (vector-update work 25,440 * 159 = 4,044,960 <= 20,000,000, matching your number), layer_cell_bound = min(25,281, 2*12,720, 25,441) = 25,281 - not 12,720 - so vector_coordinate_bound = 25,281 * 159 = 4,019,679 > 4,000,000 and the request is rejected raising exactly the asserted "intermediate vector-coordinate bound" error; at m=158 it charges 158^3 = 3,944,312 <= 4,000,000 and is accepted. The test therefore already pins the actual coordinate-bound crossover (158 accepted / 159 rejected). Verified directly: constructing both requests reproduces rejection only at 159 with that exact message, and the suite passes standalone, as a full file, and lane-style (`uv run --locked pytest tests/math/regular_languages -n 4 --dist worksteal --timeout=120 -m "not exhaustive"`: 43 passed). The proposed 199/200 crossover follows from substituting `profile_composition_bound` alone for `layer_cell_bound`; under the implementation 199 is rejected too (199^3 > 4,000,000), so moving the boundary there would assert acceptance of requests the envelope rejects. No code change required; resolving.
